packages/util/src/signature.ts

```diff
@@ -39,6 +39,21 @@ function calculateSigRecovery(v: bigint, chainId?: bigint): bigint {
   return v - (chainId * BigInt(2) + BigInt(35))
 }
 
+export function calculateSigRecoveryFromV(v: bigint) {
+  if (v === BigInt(0) || v === BigInt(1)) {
+    return v
+  } else {
+    const _v = Number(v)
+    if (Number.isInteger((0 - 35 - _v) / -2)) {
+      return BigInt(0)
+    } else if (Number.isInteger((1 - 35 - _v) / -2)) {
+      return BigInt(1)
+    } else {
+      throw new Error('Invalid v value')
```

Comment:
Can we here remain in the BigInt context? `v` *can* theoretically (actually: this also already happened practically) get in the > MAX_SAFE_INTEGER range, when there is a very large chain ID (we once had this for a testnet), so arithmetics here should take place in BigInt.

Just tested, there is the modulo operator `%` also working in `BigInt`, so: `BigInt(5) % BigInt(2)` gave me `1n`, so it should be possible with something like this to get this working in plain `BigInt` as well.